Source/Api/Controllers/StackController.cs

```diff
@@ -499,23 +501,23 @@ public class StackController : RepositoryApiController<IStackRepository, Stack,
             if (skip > MAXIMUM_SKIP)
                 return Ok(new object[0]);
 
-            var validationResult = QueryProcessor.Process(userFilter);
-            if (!validationResult.IsValid)
-                return BadRequest(validationResult.Message);
-
-            if (String.IsNullOrEmpty(systemFilter))
-                systemFilter = await GetAssociatedOrganizationsFilterAsync(_organizationRepository, validationResult.UsesPremiumFeatures, HasOrganizationOrProjectFilter(userFilter), "last");
+            var pr = QueryProcessor.Process(userFilter);
+            if (!pr.IsValid)
+                return BadRequest(pr.Message);
 
+            var organizations = await GetAssociatedOrganizationsAsync(_organizationRepository);
+            var ti = GetTimeInfo(time, offset, organizations.GetRetentionUtcCutoff());
+            systemFilter = String.Join(" ", new[] { systemFilter, BuildSystemFilter(organizations, userFilter, pr.UsesPremiumFeatures, "last") }.Where(f => !String.IsNullOrWhiteSpace(f)));
+            
             var sortBy = GetSort(sort);
-            var timeInfo = GetTimeInfo(time, offset);
             var options = new PagingOptions { Page = page, Limit = limit };
             
             try {
-                var results = await _repository.GetByFilterAsync(systemFilter, userFilter, sortBy, timeInfo.Field, timeInfo.UtcRange.Start, timeInfo.UtcRange.End, options);
+                var results = await _repository.GetByFilterAsync(systemFilter, userFilter, sortBy, ti.Field, ti.UtcRange.Start, ti.UtcRange.End, options);
 
-                var stacks = results.Documents.Select(s => s.ApplyOffset(timeInfo.Offset)).ToList();
+                var stacks = results.Documents.Select(s => s.ApplyOffset(ti.Offset)).ToList();
                 if (!String.IsNullOrEmpty(mode) && String.Equals(mode, "summary", StringComparison.OrdinalIgnoreCase))
-                    return OkWithResourceLinks(await GetStackSummariesAsync(stacks, timeInfo.Offset, timeInfo.UtcRange.UtcStart, timeInfo.UtcRange.UtcEnd), results.HasMore && !NextPageExceedsSkipLimit(page, limit), page);
+                    return OkWithResourceLinks(await GetStackSummariesAsync(stacks, ti.Offset, systemFilter, ti.UtcRange.Start, ti.UtcRange.End), results.HasMore && !NextPageExceedsSkipLimit(page, limit), page);
```

Comment:
@ejsmith this is really the only thing that changed and it was inconsistent with all other usages.